File: C1/q1-1.cpp
#include <iostream>
#include <string>
using namespace std;
class DetermineString
{
    string str;
public:
    DetermineString(string str) {this->str = str;};
    bool DetermineUnique()
    {
        char letters[256] = {};
        for (int i = 0; i < str.length(); i++)
        {
            int j = (int)str[i];
            if (letters[j]) return false;
            else letters[j] = true;
        }
        return true;
    };
};
int main()
{
    string s0 ("lqwerqtyu.");
    DetermineString d0(s0);
    cout << d0.DetermineUnique() << endl;
    return 0;    
}

File: C1/q1-1.py
import sys

def determine_string(str):
    """
    Read the string from input and decide if it has unique letters
    """
    letters = [None]*256
    for c in str:
        print ord(c)
        if letters[ord(c)]:
            return False                        
        else:
            letters[ord(c)] = True  
    return True
 
def main():
    string = sys.argv[1]
    print determine_string(string)

if __name__ == "__main__":
    main()

